wagtail/documents/wagtail_hooks.py

```diff
@@ -111,7 +111,9 @@ def get_context_data(self, parent_context):
         site_name = get_site_for_user(self.request.user)["site_name"]
 
         return {
-            "total_docs": get_document_model().objects.count(),
+            "total_docs": permission_policy.instances_user_has_any_permission_for(
+                self.request.user, {"add", "change", "delete"}
```

Comment:
As per the previous discussion on #5129 and https://github.com/wagtail/wagtail/issues/7958#issuecomment-1036053269, I think this should include `choose` permission as well. That way, if there's a large number of docs/images available for an editor to use, but they only have edit access over a small fraction of them, it won't under-report the count. (It does mean that the set they see when clicking through to the Images / Documents area will be smaller than reported, since that listing _does_ only show editable items - but better to give the correct-but-unintuitive number rather than misrepresenting things, I think.)